package/cloudshell/email/email_service.py

```diff
@@ -105,23 +137,30 @@ def _send(self, to_address, subject, message, cc=None):
         mess = MIMEText(message, 'html')
         msg.attach(mess)
 
+        smtp = self._login()
+
         try:
-            smtp = smtplib.SMTP(
-                host=self._email_config.smtp_server,
-                port=self._email_config.smtp_port
-            )
-            smtp.ehlo()
-            smtp.starttls()
-            smtp.login(self._email_config.user, self._email_config.password)
             smtp.sendmail(
                 from_addr=from_address,
                 to_addrs=[to_address, cc] if cc else to_address,
                 msg=msg.as_string()
             )
-            smtp.close()
-        except Exception:
-            self._logger.exception(f'Failed to send email to {to_address}')
-            raise Exception(f'Failed to send email to {to_address}')
+        except smtplib.SMTPRecipientsRefused:
+            self._logger.exception('Failed to send email: All recipients were refused.')
+            raise
+        except smtplib.SMTPSenderRefused:
+            self._logger.exception('Failed to send email: The server didn’t accept the from_addr.')
+            raise
+        except smtplib.SMTPDataError:
+            self._logger.exception('Failed to send email: The server replied with an unexpected '
+                                   'error code.')
+            raise
+        except smtplib.SMTPNotSupportedError:
+            self._logger.exception('Failed to send email: SMTPUTF8 was given in the mail_options '
+                                   'but is not supported by the server.')
+            raise
+
+        smtp.close()
```

Comment:
If we have an open connection I think we would want to close this connection also if there is an error sending out an email. So I would recommend putting "smtp.close()" in a "finally" statement

Reply:
added smtp.close() in a "finally" statement